src/auth/user.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+import mongoose, { models, SchemaType } from 'mongoose';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+
+
+const Schema = mongoose.Schema;
+
+const userSchema = new Schema({
+  //username is an email address
+  username: { type: String, required: true },
+  password: { type: String, required: true },
+  coach: { type: Schema.Types.ObjectId, ref: 'profiles' },
+  player: { type: Schema.Types.ObjectId, ref: 'profiles' },
+  team: { type: Schema.Types.ObjectId, ref: 'teams' },
+  stats: { type: Schema.Types.ObjectId, ref: 'stats' },
+  role: { type: String, enum: ['coach', 'player'] },
+
+});
+
+userSchema.pre('save', function (next) {
+  this.hashPassword(5, next);
```

Comment:
In the future, pull '5' out of this line and put it's value in a global variable. Something like 'SALT = 5' and then replace 5 here with SALT.